ocs_ci/ocs/hsbench.py

```diff
@@ -0,0 +1,192 @@
+        self.rgw_pod = pod.get_rgw_pods()
+        if self.rgw_pod:
+            # Create service account
+            self.sa_name = helpers.create_serviceaccount(self.namespace)
+            self.sa_name = self.sa_name.name
+            helpers.add_scc_policy(
+                sa_name=self.sa_name, namespace=self.namespace
+            )
+
+            # Create test pvc+pod
+            log.info(f"Create Fedora dc pod for testing in {self.namespace}")
+            pvc_size = '50Gi'
+            node_name = 'compute-0'
+            self.pod_name = 'hsbench-pod'
+            self.pvc_obj = helpers.create_pvc(sc_name=constants.DEFAULT_STORAGECLASS_RBD,
+                                              namespace=self.namespace, size=pvc_size
+                                              )
+            self.pod_obj = helpers.create_pod(constants.CEPHBLOCKPOOL, namespace=self.namespace,
+                                              pod_name=self.pod_name, pvc_name=self.pvc_obj.name,
+                                              node_name=node_name, sa_name=self.sa_name,
+                                              pod_dict_path=self.pod_dic_path, dc_deployment=True,
+                                              deploy_pod_status=constants.STATUS_COMPLETED
+                                              )
+        else:
+            raise UnexpectedBehaviour("This cluster doesn't have RGW pod(s) to perform hsbench")
+
+    def install_hsbench(self, timeout=1800):
+        """
+        Install HotSauce S3 benchmark:
+        https://github.com/markhpc/hsbench
+
+        """
+        # Install hsbench
+        log.info(f"Installing hsbench S3 benchmark on testing pod {self.pod_obj.name}")
+        hsbench_path = "github.com/markhpc/hsbench"
+        self.pod_obj.exec_cmd_on_pod("yum update -y", timeout=timeout)
+        self.pod_obj.exec_cmd_on_pod("yum install git -y", timeout=timeout)
+        self.pod_obj.exec_cmd_on_pod("yum install golang-bin -y", timeout=timeout)
+        self.pod_obj.exec_cmd_on_pod(f"go get -v {hsbench_path}", timeout=timeout)
+        log.info("Successfully installing hsbench benchmark")
+
+    def create_test_user(self):
+        """
+        Create a radosgw test user for S3 access
+
+        """
+        # Create RWG test user
+        self.toolbox = pod.get_ceph_tools_pod()
+        self.uid = self.hsbench_cr['uid']
+        display_name = self.hsbench_cr['display_name']
+        email = self.hsbench_cr['email']
+        self.access_key = self.hsbench_cr['access_key']
+        self.secret_key = self.hsbench_cr['secret_key']
+
+        log.info(f"Create RGW test user {self.uid}")
+        self.toolbox.exec_cmd_on_pod(
+            f"radosgw-admin user create --uid={self.uid} "
+            f"--display-name={display_name} --email={email} "
+            f"--access-key={self.access_key} --secret-key={self.secret_key}"
+        )
+        return
+
+    def run_hsbench(self, num_obj=None, timeout=None):
```

Comment:
I see that we create one pod and run IO through this one pod to create 1 million objects. This is a good start. Does hsbench allow this to be run through multiple clients (as in typical use cases)?

Reply:
For now, this test case is mainly for single bucket testing.  It can be run on multiple clients to distribute IOs across numbers of buckets.